modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/bindings/GMLCoordinatesTypeBinding.java

```diff
@@ -131,13 +131,23 @@ public Object parse(ElementInstance instance, Node node, Object value)
         CoordinateSequence seq = null;
         int i = 0;
         int ncoords = tuples.countTokens(); //number of coordinates
+        if(cs.equals(ts)) {
+            ncoords = ncoords/2;
```

Comment:
Yes, you're correct, it does assume 2D, but I figured: it was better than before:-).
However, I might be wrong, but in my tests I think I encountered GML3 geometries, which were parsed with this binding.